docs/changelog/130026.yaml

```diff
@@ -0,0 +1,6 @@
+pr: 130026
+summary: Ensure date/date_nanos implicit casting rule behind snapshot
+area: ES|QL
+type: bug
```

Comment:
This re-appeared - I think you'll have to mark this PR as `>non-issue`, otherwise the bot will probably insist on the changelog entry.